torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,446 @@
+
+    reference_corpus = [[preprocess_function(ref) for ref in reference] for reference in reference_corpus]
+    hypothesis_corpus = [preprocess_function(hyp) for hyp in hypothesis_corpus]
+
+    return reference_corpus, hypothesis_corpus
+
+
+def _compute_sentence_statistics(
+    reference_words: List[List[str]],
+    hypothesis: List[str],
+    alpha: float = 2.0,
+    rho: float = 0.3,
+    deletion: float = 0.2,
+    insertion: float = 1.0,
+) -> Tuple[Tensor, Tensor]:
+    """Compute scores for EED.
+
+    Args:
+        reference_words:
+            An iterable of reference words
+        hypothesis:
+            A sentence
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
+        rho:
+            coverage cost, penalty for repetition of characters
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+
+    Returns:
+        Tuple of scores and number of sentences as floats
+    """
+    scores = 0.0
+    num_sentences = 0.0
+
+    for reference in reference_words:
+        score = _eed_function(reference, hypothesis, alpha, rho, deletion, insertion)
+        scores += score
+        num_sentences += 1.0
+
+    return scores, num_sentences
+
+
+def _eed_update(
+    reference_corpus: Sequence[Union[str, Sequence[str]]],
+    hypothesis_corpus: Union[str, Sequence[str]],
+    language: Literal["en", "ja"] = "en",
+    return_sentence_level_score: bool = False,
+    alpha: float = 2.0,
+    rho: float = 0.3,
+    deletion: float = 0.2,
+    insertion: float = 1.0,
+) -> Tuple[Tensor, Tensor]:
+    """Compute scores for EED.
+
+    Args:
+        reference_corpus:
+            An iterable of iterables of reference corpus
+        hypothesis_corpus:
+            An iterable of hypothesis corpus
+        language:
+            Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+        return_sentence_level_score:
+            An indication of whether sentence-level EED is to be returned
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
+        rho:
+            coverage cost, penalty for repetition of characters
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+
+    Returns:
+        Tuple of scores, total sentences as floats, and individual sentence scores
+    """
+    reference_corpus, hypothesis_corpus = _preprocess_sentences(reference_corpus, hypothesis_corpus, language)
+
+    # check if reference_corpus or hypothesis_corpus is empty
+    if 0 in (len(hypothesis_corpus), len(reference_corpus[0])):
+        scores = tensor(0.0)
+        total_num_sentences = tensor(0.0)
+        sentence_eed = []
+        return scores, total_num_sentences, sentence_eed
```

Comment:
I would simplify these and store only sentence-level EED. From this, you can then calculate average EED without storing `total_num_sentences`

Reply:
adding on next commit